### PR TITLE
[rfw] fix macos debug entitlements for wasm example

### PR DESCRIPTION
**Pull request by dkbast**
The macos version of the rfw wasm example was missing debug entitlements to connect to the internet and fetch the remote file. 

This is related to https://github.com/flutter/flutter/issues/90218

I belive no changes to `CHANGELOG.md` or  `pubspec.yaml` are necessary as this does not affect the pub release.

Tagging @Hixie for review.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: testing this would require running the example on a mac host and doing network traffic, which would be unfortunately rather flake-prone.

**Comment by Hixie** [comment]
Might be worth adding a CHANGELOG entry just for people to know we changed something.

File: packages/rfw/example/wasm/macos/Runner/DebugProfile.entitlements
Comment (Hixie) [inline packages/rfw/example/wasm/macos/Runner/DebugProfile.entitlements:12]: should we add this to the other entitlements too? (notably the release mode one)

**Review by Hixie — APPROVED** [review]
LGTM

(This will need a second reviewer before it can land.)

File: packages/rfw/example/wasm/macos/Runner/DebugProfile.entitlements
Comment (stuartmorgan) [reply packages/rfw/example/wasm/macos/Runner/DebugProfile.entitlements:12]: This should definitely be in the release entitlements as well.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
@dkbast Are you planning on updating this based on my comment above?

**Comment by dkbast** [comment]
Added it to the release entitlements as well.

The example itself is still not running under M1 mac, as reported here: https://github.com/flutter/flutter/issues/90218

**Review by stuartmorgan — APPROVED** [review]
LGTM, thanks!

**Comment by stuartmorgan** [comment]
Override: running the example is a dev-only change, so exempt from changelog and version changes.

**Pull request merged**
